Automattic/jetpack
# Blaze: add new dashboard menu

## Proposed changes:

This registers a new "Tools > Advertising" menu, appearing when your site is eligible for Blaze.

The contents of the menu will be loaded remotely from Calypso.

This borrows from the Stats Admin package, where we've implemented something similar (the Stats page is loaded from Calypso).

**To do**

- [x] Figure out if the menu location should change in different environments (e.g. sites that do not use the Jetpack plugin, but may require Blaze on its own). peeHDf-Zg-p2
- [x] Maybe add gating so the menu is not publicly accessible until this is ready for prime-time.
- [x] Figure out if this menu should be hidden or handled differently on WordPress.com sites or WoA sites, where we already have the Tools > Advertising menu. peeHDf-Zg-p2

### Other information:


## Jetpack product discussion

* https://github.com/Automattic/wp-calypso/pull/76431
* pedfti-1M1-p2
* Related: pbIy4N-3vY-p2
* p1681412672232459-slack-C053FAERCDP
* peeHDf-Zg-p2
* D112493-code

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

1. Ensure that Blaze links still appear under the published posts in Posts > All Posts, and that the links work.
2. Ensure that you're still offered to Blaze a post in the post-publish panel in the editor (after publishing a post), and that the Blaze link properly redirects you to the Advertising dashboard in Calypso.
3. Add `add_filter( 'jetpack_blaze_dashboard_enable', '__return_true' );` to a functionality plugin on your site.
4. You should see a new Tools > Advertising menu when your site is connected to WordPress.com.

*******************
To populate the contents of the dashboard, you have 2 options:

* You can patch D112493-code in your WordPress.com sandbox.
* You can build your own copy of the dashboard:
    * Build Blaze and Jetpack: `jetpack build packages/blaze && jetpack build plugins/jetpack`
    * Clone https://github.com/Automattic/wp-calypso
    * Install deps with `yarn`
    * Go to `apps/blaze-dashboard`
    * Launch the app with `BLAZE_DASHBOARD_PACKAGE_PATH=/.../jetpack/projects/packages/blaze yarn dev`, while entering the correct path to your package on your machine.
*******************

5. Visit the Tools > Advertising menu on your local installation.
6. Now re-test items 1 and 2 of this list; they should still work, but the links should now point to the wp-admin dashboard.
